CMakeLists.txt

```diff
@@ -95,3 +95,74 @@ function(trusted_firmware_build)
   endif()
 
 endfunction()
+
+# Gets a list of commands to be performed after a successful Zephyr build that
+# involves trusted-firmware-m.
+function(trusted_firmware_get_post_build_commands out_commands)
+
+  if(NOT CONFIG_TFM_BL2_FALSE)
+
+    # Set default image versions if not defined elsewhere
+    if(NOT DEFINED TFM_IMAGE_VERSION_S)
+      set(TFM_IMAGE_VERSION_S 0.0.0+0)
+    endif()
+    if(NOT DEFINED TFM_IMAGE_VERSION_NS)
+      set(TFM_IMAGE_VERSION_NS 0.0.0+0)
+    endif()
+
+    set(TFM_MCUBOOT_DIR "${ZEPHYR_TFM_MODULE_DIR}/trusted-firmware-m/bl2/ext/mcuboot")
+    set(PREPROCESSED_FILE "${CMAKE_BINARY_DIR}/tfm/image_macros_preprocessed")
+
+    if(NOT DEFINED TFM_PUBLIC_KEY_FORMAT)
+      set(TFM_PUBLIC_KEY_FORMAT "full")
+    endif()
+
+    set(${out_commands}
+
+      # Sign secure binary image with public key
+      COMMAND ${PYTHON_EXECUTABLE} ${TFM_MCUBOOT_DIR}/scripts/imgtool.py
+        sign
+        --layout ${PREPROCESSED_FILE}_s.c
+        -k ${CONFIG_TFM_KEY_FILE_S}
+        --public-key-format ${TFM_PUBLIC_KEY_FORMAT}
+        --align 1
+        -v ${TFM_IMAGE_VERSION_S}
+        ${ADD_NS_IMAGE_MIN_VER}
+        ${ADD_SECURITY_COUNTER_S}
```

Comment:
Those two variables may not be set, that depends on current variables in the scope of the caller, and may lead to unexpected behavior depending on the location from where this function is called.

Probably this should be arguments to the function.

Reply:
This was copied from the mps2_an521 board's [CMakeLists.txt](https://github.com/zephyrproject-rtos/zephyr/blob/12e1fd653deab52cef5b1679ba631c35b2a55286/boards/arm/mps2_an521/CMakeLists.txt#L45-L46). Looking at it now, I don't know how it was supposed to work. Perhaps I should add here:
```
    if (TFM_NS_IMAGE_MIN_VER)
      set(ADD_NS_IMAGE_MIN_VER "-d \"(1,${TFM_NS_IMAGE_MIN_VER})\"")
    else()
      set(ADD_NS_IMAGE_MIN_VER "")
    endif()
    if (TFM_SECURITY_COUNTER_S)
      set(ADD_SECURITY_COUNTER_S "-s ${TFM_SECURITY_COUNTER_S}")
    else()
      set(ADD_SECURITY_COUNTER_S "")
    endif()
```
basing on what is done in MCUBoot.cmake: https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/143df675557305b61f7930a50459a53a8d2bb097/trusted-firmware-m/bl2/ext/mcuboot/MCUBoot.cmake#L127-L131
https://github.com/zephyrproject-rtos/trusted-firmware-m/blob/143df675557305b61f7930a50459a53a8d2bb097/trusted-firmware-m/bl2/ext/mcuboot/MCUBoot.cmake#L92-L96

Reply:
@microbuilder you took over from @karl-zh, do you have any comments on this ?

Reply:
For reference: https://github.com/zephyrproject-rtos/zephyr/blame/12e1fd653deab52cef5b1679ba631c35b2a55286/boards/arm/mps2_an521/CMakeLists.txt#L45-L46